source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -143,8 +146,52 @@ void TextureBaker::bakeGraphOutput(OutputPtr output, GenContext& context, const
     ShaderPtr shader = _generator->generate("BakingShader", output, context);
     createProgram(shader);
 
+    if (_autoTextureResolution)
+    {
+        GlslProgramPtr program = getProgram();
+        GLFrameBufferPtr framebuffer = getFrameBuffer();
+        unsigned int bakedTextureHeight = 0;
+        unsigned int bakedTextureWidth = 0;
+        bool requiresResize = false;
+        // Prefetch all required images and query thier dimensions. 
+        // Since Images are cached by ImageHandler, they will be reused during bindTextures
+        const GlslProgram::InputMap& uniformList = program->getUniformsList();
+        for (const auto& uniform : uniformList)
+        {
+            GLenum uniformType = uniform.second->gltype;
```

Comment:
I can look this. I kept the OpenGL types since this was in MaterialXRenderGlsl

Reply:
That would be a great help, thanks @ashwinbhat.  If there was already a dependency on glew.h in TextureBaker.cpp, I wouldn't see any harm in referencing it, but it seems like a step backwards to introduce a new OpenGL dependency that isn't already there.  The TextureBaker class seems particularly amenable to generalizing beyond OpenGL in the future, and even within OpenGL it's best to minimize our dependencies on specific support libraries.

Reply:
Material::bindImages() has an example of non language specific public uniform traversal. 
 A GLProgram creation actually does a remapping from Shader to OpenGL specific constructs so it's 1:1.
 
 As for generalizing, yes, this is something we would want as well. I think removing TextureBaker from the class hierarchy of a renderer would be the first step so a baker uses a renderer and is not a rendere itself. This would remove easy access to the GLProgram as a start. For afterwards :).

Reply:
@bernardkwok  I tried to remove the OpenGL dependency [here](https://github.com/autodesk-forks/MaterialX/blob/cba0639e3782500a73f691e2c78b64ec8eb331ee/source/MaterialXRenderGlsl/TextureBaker.cpp#L157-L174)
The "shader" reports unused textures but the "glProgram" report the textures that will indeed be used for the current output.

Reply:
@jstone-lucasfilm , Ashwin and I had a talk and we can virtualize the logic to for this and have a specific OpenGL logic version.

Note that due to OpenGL program optimization the exposed uniforms on the MaterialX Shader don't necessarily map those on the GLProgram. Thus there can be additional time to load in an images to find it's size when it is not an active uniform being used. The GL specific virtual method will user the optimized set of uniforms as is currently shonw. But if/when we try to bake using OSL, we may need to test more images as the compiled OSL program would not be parsed.